test/jasmine/tests/legend_test.js

```diff
@@ -628,3 +629,188 @@ describe('legend restyle update', function() {
+                expect(+legendItem.parentNode.style.opacity).toBeLessThan(1);
+            });
+            it('should unhide series', function(done) {
+                legendItem.dispatchEvent(new MouseEvent('mousedown'));
+                legendItem.dispatchEvent(new MouseEvent('mouseup'));
+                setTimeout(function() {
+                    expect(gd.data[2].visible).toBe(true);
+                    done();
+                }, DBLCLICKDELAY * 2);
+            });
+            it('should unfade legend item', function() {
+                expect(+legendItem.parentNode.style.opacity).toBe(1);
+            });
+        });
+        describe('double click', function() {
+            it('should hide series', function(done) {
+                legendItem.dispatchEvent(new MouseEvent('mousedown'));
+                legendItem.dispatchEvent(new MouseEvent('mouseup'));
+                legendItem.dispatchEvent(new MouseEvent('mousedown'));
+                legendItem.dispatchEvent(new MouseEvent('mouseup'));
+                setTimeout(function() {
+                    for(var i = 0; i < legendItems.length; i++) {
+                        if(i === testEntry) {
+                            expect(gd.data[i].visible).toBe(true);
+                        } else {
+                            expect(gd.data[i].visible).toBe('legendonly');
+                        }
+                    }
+                    done();
+                }, DBLCLICKDELAY);
+            });
+            it('should fade legend item', function() {
+                var legendItemi;
+                for(var i = 0; i < legendItems.length; i++) {
+                    legendItemi = legendItems[i];
+                    if(i === testEntry) {
+                        expect(+legendItemi.parentNode.style.opacity).toBe(1);
+                    } else {
+                        expect(+legendItemi.parentNode.style.opacity).toBeLessThan(1);
+                    }
+                }
+            });
+            it('should unhide series', function(done) {
+                legendItem.dispatchEvent(new MouseEvent('mousedown'));
+                legendItem.dispatchEvent(new MouseEvent('mouseup'));
+                legendItem.dispatchEvent(new MouseEvent('mousedown'));
+                legendItem.dispatchEvent(new MouseEvent('mouseup'));
+                setTimeout(function() {
+                    for(var i = 0; i < legendItems.length; i++) {
+                        expect(gd.data[i].visible).toBe(true);
+                    }
+                    done();
+                }, DBLCLICKDELAY);
+            });
+            it('should unfade legend items', function() {
+                var legendItemi;
+                for(var i = 0; i < legendItems.length; i++) {
+                    legendItemi = legendItems[i];
+                    expect(+legendItemi.parentNode.style.opacity).toBe(1);
+                }
+            });
+        });
+    });
+});
```

Comment:
These tests are great, and they lock down a lot more behavior than just what you added, thanks! 🍻 

Can we reduce the delays a bit? I guess the `DBLCLICKDELAY * 2` delays just need to be `> DBLCLICKDELAY`, can we drop them to `DBLCLICKDELAY + 20` or something? And the delays after a double click can presumably be very short (perhaps `20`)?

Non-blocking, but you might also want to make helpers to reduce repetition generating the mouse events. Either `click(element)` and `doubleclick(element)` or `click(element, count)`.

Oh, but this one *is* blocking: we have [click](https://github.com/plotly/plotly.js/blob/master/test/jasmine/assets/click.js) and [double_click](https://github.com/plotly/plotly.js/blob/master/test/jasmine/assets/double_click.js) helpers in test assets - which aren't useful here because they're linked to mouse coordinates rather than an element - but I see that double_click uses `DBLCLICKDELAY` so needs its reference updated. I guess the tests that use it will still pass with no delay, but not exactly what we wanted them to be testing...